front/lib/resources/group_resource.ts

```diff
@@ -0,0 +1,134 @@
+import type { ModelId, Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type { Attributes, ModelStatic, Transaction } from "sequelize";
+
+import type { Authenticator } from "@app/lib/auth";
+import type { Workspace } from "@app/lib/models/workspace";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { GroupModel } from "@app/lib/resources/storage/models/groups";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+import { getResourceIdFromSId, makeSId } from "@app/lib/resources/string_ids";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unsafe-declaration-merging
+export interface GroupResource extends ReadonlyAttributesType<GroupModel> {}
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class GroupResource extends BaseResource<GroupModel> {
+  static model: ModelStatic<GroupModel> = GroupModel;
+
+  private workspace?: Workspace;
+
+  constructor(model: ModelStatic<GroupModel>, blob: Attributes<GroupModel>) {
+    super(GroupModel, blob);
+  }
+
+  static async makeNew(blob: Attributes<GroupModel>) {
+    const group = await GroupModel.create({
+      ...blob,
+    });
+
+    return new this(GroupModel, group.get());
+  }
+
+  get sId(): string {
+    return GroupResource.modelIdToSId({
+      id: this.id,
+      workspaceId: this.workspaceId,
+    });
+  }
+
+  static modelIdToSId({
+    id,
+    workspaceId,
+  }: {
+    id: ModelId;
+    workspaceId: ModelId;
+  }): string {
+    return makeSId("group", {
+      id,
+      workspaceId,
+    });
+  }
+
+  async update(
```

Comment:
Do we need this. We prefer specific updates vs a generic one? Is it required by the BaseResource?

Reply:
I guess will only need to update the name & isWorkspace? Do you prefer on method for each?

Reply:
I'll simply remove it and we can add if if and when we need it